gpii/node_modules/gpii-db-operation/src/DbDataStoreUtils.js

```diff
@@ -61,19 +62,33 @@ gpii.dbOperation.dbDataStore.findRecord = function (dataSource, directModel, val
             // by a document id, 404 status is returned and this conversion is not needed.
             var result = $.isEmptyObject(data) ? undefined : dataProcessFunc(data);
             if (result !== undefined && result.isError) {
+                result.dbUrl = gpii.dbOperation.dbDataStore.sanitizeUrl(dataSource);
                 promiseTogo.reject(result);
             } else {
                 promiseTogo.resolve(result);
             }
         }, function (error) {
+            error.dbUrl = gpii.dbOperation.dbDataStore.sanitizeUrl(dataSource);
             fluid.log("gpii-db-operation, findRecord(), error occurs: ", error);
             promiseTogo.reject(error);
         });
     }
-
     return promiseTogo;
 };
 
+/**
+ * Return a version of the database URL without showing senstive information,
+ * such as user name and/or password.
+ * @param {Component} dataSource - An instance of gpii.dbOperation.dbDataSource.
+ * @return {String} A sanatized version of the url.
+ */
+gpii.dbOperation.dbDataStore.sanitizeUrl = function (dataSource) {
```

Comment:
When this kettle pull request https://github.com/fluid-project/kettle/pull/49 is merged, the censoring of the sensitive information will be handled at the lower level in kettle. 

It should be safe to directly return messages from kettle.

@amb26, can you confirm if this comment makes sense? Thanks.

Reply:
@cindyli - yes, that is correct - although the improvements in Kettle will only affect its own logging of the URLs. Given we independently log the URL in error messages here, it is reasonable to perform our own sanitisation.

Reply:
The need for removing sensitive information before reporting URLs has come up before (e.g., deleting and loading the snapsets).  It sounds like there is a need for a general function somewhere.  Is Infusion too low level a place to put this (`fluid.censorUrlCredentials()`)? If that's too low, maybe a function in the `gpii` namespace.